blocks/user-licenses/user-licenses.js

```diff
@@ -1,5 +1,400 @@
+  addUserActionButton.setAttribute('contactId', event.target.getAttribute('id'));
+}
+
+async function addUserToALicense(event) {
+  const contactId = event.target.getAttribute('contactid');
+  const licenseId = event.target.getAttribute('data-license-id');
+  const data = await execute('dynamics_add_end_user', `&contactId=${contactId}&licenseid=${licenseId}`, 'POST');
+
+  if (data !== undefined && data !== null && data.status === 204) {
+    // TODO show success toast message
+    hideModal(event);
+    // eslint-disable-next-line no-use-before-define
+    displayLicenseDetails(event);
+  } else {
+    console.log('error ', data);
+  }
+}
+
+async function createColleaguesTable(checkboxClass) {
+  const data = await execute('dynamics_get_colleagues_view', '', 'GET');
+  const headingsMapping = [
+    {
+      label: '',
+      value: [],
+      html: 'input',
+      htmlAttributes: {
+        class: `${checkboxClass}`,
+        type: 'checkbox',
+        id: '{{contactid}}',
+      },
+    },
+    {
+      label: 'Full Name',
+      value: ['{{firstname}}', ' ', '{{lastname}}'],
+    },
+    {
+      label: 'Job Title',
+      value: ['{{jobtitle}}'],
+    },
+    {
+      label: 'Email',
+      value: ['{{emailaddress1}}'],
+    },
+    {
+      label: 'Business Phone',
+      value: ['{{telephone1}}'],
+    },
+  ];
+  return createGenericTable(headingsMapping, data.colleagues);
+}
+
+async function addColleaguesToUserActionModal(
+  modalSelectorClass,
+  modalCheckBoxClass,
+  eventListenerMethoda,
+) {
+  const modalBodyDivs = document.querySelectorAll('.modal-body');
+  let modalContentDiv = null;
+  modalBodyDivs.forEach((modalBodyDiv) => {
+    const containerDiv = modalBodyDiv.querySelector(`.${modalSelectorClass}`);
+    if (containerDiv !== null && containerDiv !== undefined) {
+      modalContentDiv = containerDiv;
+    }
+  });
+
+  // clear previous modal content
+  modalContentDiv.innerHTML = '';
+  modalContentDiv.appendChild(await createColleaguesTable(modalCheckBoxClass));
+  const userCheckboxes = document.querySelectorAll(`.${modalCheckBoxClass}`);
+  userCheckboxes.forEach((userCheckbox) => {
+    userCheckbox.addEventListener('click', eventListenerMethoda);
+  });
+}
+
+async function removeUserFromLicense(event) {
+  const newproductuserid = event.target.getAttribute('data-new-productuserid');
+  const data = await execute('dynamics_remove_enduser_from_license', `&new_productuserid=${newproductuserid}`, 'DELETE');
+  // TODO handle response and add toast message
+
+  if (data !== undefined && data !== null && data.status === 204) {
+    // TODO show success toast message
+    hideModal(event);
+    // eslint-disable-next-line no-use-before-define
+    displayLicenseDetails(event);
+  } else {
+    console.log('error ', data);
+  }
+}
+
+async function updateLicenseNickname() {
+  const { value } = document.querySelector('.nickname');
+  if (value !== '' && value !== undefined) {
```

Comment:
```suggestion
  if (value) {
```
This code will execute the block of code inside the if statement if value is truthy, which means it is not undefined, null, an empty string (''), NaN, 0, or false.